### ISSUE TITLE
no-unreachable false positive

### ISSUE THREAD

**Issue by lucacasonato** [denoland/deno_lint#477]
```
    const partsA = [];
    const partsB = [];
    for (let i = 0; i < Math.max(partsA.length, partsB.length); i++) {
      const partA = partsA[i];
      const partB = partsB[i];
      if (partA === undefined) return -1;
      if (partB === undefined) return 1;
      if (partA === partB) continue;
      const priorityA = partA.startsWith(":") ? partA.endsWith("*") ? 0 : 1 : 2;
      const priorityB = partB.startsWith(":") ? partB.endsWith("*") ? 0 : 1 : 2;
      return Math.max(Math.min(priorityB - priorityA, 1), -1);
    }
    return 0;
```

Here the `return 0;` is marked as unreachable, although it is reachable if partsA.length == 0 & partsB.length == 0.

**Comment by qwerasd205** [comment]
Minimal reproduction (that also shows a missed case, the interior of the for loop is unreachable)
```js
function test () {
    for (;false;) {
        return 0;
    }
    return 1;
}
```

**Comment by magurotuna** [comment]
I'll fix it
